src/main/java/ua/bank/bankservice/util/CurrencyConverter.java

```diff
@@ -0,0 +1,33 @@
+package ua.bank.bankservice.util;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import org.springframework.stereotype.Component;
+import ua.bank.bankservice.model.Currency;
+
+@Component
+public class CurrencyConverter {
+    public double convert(Currency fromCurrency, Currency toCurrency, int amount) {
+        String urlStr = "https://api.exchangerate.host/convert?from=" + fromCurrency.name()
+                + "&to=" + toCurrency.name() + "&amount=" + amount;
+        try {
+            URL url = new URL(urlStr);
+            HttpURLConnection request = (HttpURLConnection) url.openConnection();
+            request.connect();
+            JsonElement root = JsonParser.parseReader(new InputStreamReader(
```

Comment:
it is not required to create a new InputStream

Reply:
Let's create a separate class `HttpClient` and path the URL as a param. In this class let's call a method from that class. See example in Thursday's or Friday's lesson.

Reply:
Remove this part from current HW